app/models/piece.rb

```diff
@@ -158,6 +139,20 @@ def is_white?(piece)
  		piece.color == "white" ? true : false
  	end	
 
+ 	def puts_in_check?(x,y)
+		if color == "white"
+			opposite_color = "black"
+		elsif color == "black"
+			opposite_color = "white"
+		end
+				
+		opponents = game.pieces.where(color: opposite_color)
+		opponents.each do |piece|
+			return true if piece.valid_move?(x,y) 
+			break
```

Comment:
Not sure why we need break here. From my understanding , the break statement will exit the loop if return true doesn't execute. So only one "opponent" attempts a move on the King. That's what I see, let me know if I'm wrong.